src/cloud-api-adaptor/test/e2e/assessment_helpers.go

```diff
@@ -218,6 +189,55 @@ func ComparePodLogString(ctx context.Context, client klient.Client, customPod v1
 	return podLogString, nil
 }
 
+// Note: there are currently two event types: Normal and Warning, so Warning includes errors
+func GetPodEventWarningDescriptions(ctx context.Context, client klient.Client, pod v1.Pod) (string, error) {
+	clientset, err := kubernetes.NewForConfig(client.RESTConfig())
+	if err != nil {
+		return "", err
+	}
+
+	events, err := clientset.CoreV1().Events(pod.Namespace).List(ctx, metav1.ListOptions{FieldSelector: fmt.Sprintf("involvedObject.name=%s", pod.Name)})
+	if err != nil {
+		return "", err
+	}
+
+	var descriptionsBuilder strings.Builder
+	for _, event := range events.Items {
+		if event.Type == v1.EventTypeWarning {
+			descriptionsBuilder.WriteString(event.Message)
+		}
+	}
+	return descriptionsBuilder.String(), nil
+}
+
+func ComparePodEventWarningDescriptions(ctx context.Context, t *testing.T, client klient.Client, pod v1.Pod, expectedPodEvent string) error {
```

Comment:
Suggestion: document the function so users understand the logic easily. You did it very well on the commit message, btw.